### PR TITLE
Pass Legend Shape prop to Overflow State

### PR DESCRIPTION
**Pull request by skyfong**
## Previous Behavior

Currently, if any of the legends props in the charts are overridden, the legends lose their hover functionality that highlights the corresponding chart data. Therefore, the user cannot change the shape of the legends and its items in overflow state without compromising legend functionality. 

OnHover/OnClick: 
image

## New Behavior

Shape prop is now passed into overflow props, so that the overflow hover card will match the legend items that are not in overflow. 
image

OnHover: 
image
image

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/react-charting/src/components/Legends/Legends.base.tsx [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
         hoverAction: legend.hoverAction!,
         onMouseOutAction: legend.onMouseOutAction!,
         color: legend.color,
-        shape: legend.shape,
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/Legends/Legends.base.tsx:106]: if shape is defined then use it else use legend.shape as a fallback

File: packages/react-charting/src/components/Legends/Legends.base.tsx [hunk 2]
```diff
@@ -88,7 +88,7 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
   }
 
   private _generateData(): ILegendOverflowData {
-    const { allowFocusOnLegends = true } = this.props;
+    const { allowFocusOnLegends = true, shape } = this.props;
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/Legends/Legends.base.tsx:91]: add this prop in ILegendsProps in Legends.types.ts as an optional property.

File: packages/react-charting/src/components/Legends/Legends.base.tsx [hunk 3]
```diff
@@ -103,7 +103,7 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
         hoverAction: legend.hoverAction!,
         onMouseOutAction: legend.onMouseOutAction!,
         color: legend.color,
-        shape: legend.shape,
+        shape: shape,
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/Legends/Legends.base.tsx:106]: you will need to generate a change file by running `yarn change` from the project root

**Comment by AtishayMsft** [comment]
are you able to validate this functionality

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by AtishayMsft** [comment]
The build is failing because the api signature has changed. Could you run a yarn build locally and add react-charting.api.md to the PR.

**Comment by skyfong** [comment]
> are you able to validate this functionality

validated in the screenshots that i just added

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**